### PR TITLE
Shot timing data

### PR DESCRIPTION
**Pull request by martijnjanssen**
This branch adds checks for adding a shot the the timelines. This branch still has unresolved checkstyle issues that need fixing.

### CODE REVIEW

**Comment by geenen124** [comment]
Looking forward to integrating this into a controller!

File: src/main/java/data/CameraTimeline.java [hunk 1]
```diff
@@ -1,31 +1,95 @@
...
 /**
- * Created by Bart.
  * Class to store information about a camera timeline.
+ * @author Bart.
  */
 public class CameraTimeline extends Timeline {
 
     // The camera that is associated with this timeline.
-    @Getter @Setter
+    @Getter
```
Comment (mennoo1996) [inline src/main/java/data/CameraTimeline.java:16]: Until now we have used @Getter and @Setter on the same line. Let's keep it the same everywhere

File: src/main/java/data/Shot.java [hunk 2]
```diff
@@ -13,17 +13,79 @@
...
+        int result = Double.compare(getStartCount(), other.getStartCount());
+
+        if (result == 0) {
+            result = Double.compare(getEndCount(), other.getEndCount());
+        }
+
+        return result;
+    }
+
+    // TODO: 28-4-16 Implement the offset due to setup of cameras
```
Comment (mennoo1996) [inline src/main/java/data/Shot.java]: Does this still have to happen?

File: src/main/java/data/Timeline.java [hunk 3]
```diff
@@ -14,13 +12,27 @@
...
     }
+
+    /**
+     * Checks overlap between two shots. If the two shots are overlapping, the two shots will have
+     * their overlapping variables set to true.
+     * @param s1 the first Shot to check overlap
+     * @param s2 the other Shot to check overlap
+     * @return true when the two shots are overlapping, false if not
+     */
+    public boolean checkOverlap(Shot s1, Shot s2) {
```
Comment (mennoo1996) [inline src/main/java/data/Timeline.java]: This method is never used. Do we still need it?

**Comment by bart-de-jonge** [comment]
Looks fine to me, great work

**Comment by geenen124** [comment]
Looks good!

**Pull request merged**